### PR TITLE
ENT-11094: Improved rpm verification for hub package

### PR DESCRIPTION
**Pull request by nickanderson**
See also: https://github.com/cfengine/masterfiles/pull/2797

### CODE REVIEW

File: packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in [hunk 1]
```diff
@@ -373,6 +374,8 @@ exit 0
 
 %defattr(460,root,cfapache,570)
 %prefix/httpd/conf
+%config %prefix/httpd/conf/httpd.conf
+%attr(400,root,root) %prefix/httpd/conf/httpd.conf
```
Comment (craigcomstock) [inline packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in:378]: does this match policy? somewhere near here? https://github.com/cfengine/masterfiles/blob/master/cfe_internal/enterprise/mission_portal.cf

Comment (nickanderson) [reply packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in:378]: The policy for this is here: <https://github.com/cfengine/masterfiles/blob/master/cfe_internal/enterprise/mission_portal.cf#L191-L203>

    "$(config)"  -> { "ENT-9686" }
      copy_from => local_dcp( $(staged_config) ),
      handle => "cfapache_httpd_conf_from_staged",
      if => and( or( "apache_stop_after_new_staged_config_repaired",
                     not( fileexists( "$(httpd_pid_file)" ) ),
                     isnewerthan( $(config), $(staged_config) ) ),
                 returnszero("$(validate_config) > /dev/null 2>&1 ", "useshell")),
      classes => results("bundle", "mission_portal_apache_config"),
      comment => "We make sure that the deployed config is a copy of the staged
                  config if the staged config passes a syntax check. We redirect
                  the output to keep policy runs clean and not generate lots of
                  unnecessary email.";

The policy currently makes no assertion for perms. So, it's defaulting to root/root.

I suppose that I could add `perms => mog( "400", "root", "root")` to that promise. Probably a good idea. Will open separate ticket/pr if you agree.

Comment (craigcomstock) [reply packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in:378]: sounds excellent. maybe a link in both to the other? aka comment in the spec with policy location (not URL necessarily) and comment in policy with spec location. as you wish. :)

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10156/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10156/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10156/

File: packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in [hunk 2]
```diff
@@ -373,6 +373,8 @@ exit 0
 
 %defattr(460,root,cfapache,570)
 %prefix/httpd/conf
+%config %prefix/httpd/conf/httpd.conf
```
Comment (vpodzime) [inline packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in:377]: This means that after upgrade there will be `httpd.conf.rpmnew` if the file doesn't match the version from the older package, right?

**Review by vpodzime — COMMENTED** [review]
~it's~ *its* in the last commit message. Otherwise this looks good to me. I didn't even know `%verify` was a thing. Very nice!

File: packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in (hunk 2 above)
Comment (nickanderson) [reply packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in:377]: Uh, I don't know. I didn't think so. I thought that happens only if you use `%config(noreplace)`.

Comment (nickanderson) [reply packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in:377]: https://rpm-software-management.github.io/rpm/manual/spec.html
```
%config

The %config(missingok) indicates that the file need not exist on the installed machine. The %config(missingok) is frequently used for files like /etc/rc.d/rc2.d/S55named where the (non-)existence of the symlink is part of the configuration in %post, and the file may need to be removed when this package is removed. This file is not required to exist at either install or uninstall time.

The %config(noreplace) indicates that the file in the package should be installed with extension .rpmnew if there is already a modified file with the same name on the installed machine.
```

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10183/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10183/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10183/

**Review by vpodzime — DISMISSED** [review]
ACK. Thanks! :clap:

**Review by craigcomstock — APPROVED** [review]

File: packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in [hunk 3]
```diff
@@ -97,6 +97,7 @@ cp -R %{_basedir}/mission-portal $RPM_BUILD_ROOT%prefix/share/GUI
 
 rm -f $RPM_BUILD_ROOT%prefix/share/GUI/runalerts.php
 cp -v %{_basedir}/mission-portal/runalerts.php $RPM_BUILD_ROOT%prefix/bin/runalerts.php
+%verify(owner group mode maj min symlink) %prefix/bin/runalerts.php
```
Comment (craigcomstock) [inline packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in:100]: looks like rhel-7 doesn't like this line? Not sure if this is the line but would seem so.

```
14:04:00 /var/tmp/rpm-tmp.D6vpyY: line 70: syntax error near unexpected token `owner'
```

Comment (nickanderson) [reply packaging/cfengine-nova-hub/cfengine-nova-hub.spec.in]: Ah, it was not in the correct section. Needed to be in `%files`, I think this change I just push fixes it.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10254/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10254/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10254/

**Comment by nickanderson** [comment]
Last build was green. Shall we merge to master?

**Review by craigcomstock — APPROVED** [review]

**Pull request merged**